### DISCUSSION TITLE
Is there a way to pass variables/files between workflows?

### DISCUSSION THREAD

**Discussion by greenaar** [woodpecker-ci/woodpecker#4364] [Ask a Question]
An example.

My build step builds and tags my docker image, then pushes it out, then sends a discord message.  All of this is currently within one workflow.

If i could pass either files or variables easily between workflows, I could break out a discord.yaml and a push.yaml.  This would reduce the size of the main workflow file, while allowing these mostly standard code blocks to be simply copied to new projects as needed.

Ideally I'd like to either share an environment variable, or a volume. Given the limited data size (usually a string or two), either would be nice -- and easier than say uploading to s3, then having to figure out what I need to download from s3 for the next step.  To be clear, this isn't trying to solve portability between steps in the same workflow, i have been following #2266 with that in mind.  I'm looking for any way to get data from one workflow to another (that is using depends_on to be run after).

I'm running in kubernetes, via helm install.

Apologies if this has been asked before.  I'm also happy to consider alternative mechanisms.

Thanks for reading this.

**Comment by qwerty287** [comment]
Looks like you're looking for something like https://github.com/woodpecker-ci/woodpecker/discussions/2272?

**Comment by zc-devs** [comment]
> share a volume

You would need `ReadWriteMany` volume. And then will probably complain again (#3869).
Problems for `ReadWriteOnce` #3344 in case you would like to chain workflows in a row.

**Comment by greenaar** [comment]
Useful information.

I suppose I'll go down the RWX PVC route for the moment. I don't love it as a permanent solution though.  Manually created PV, then the efforts required to avoid race conditions or otherwise stomping on anything else that might be mounting that volume.

#2272 looks like a better long term solution, but based on the last update, it's probably a longer wait.

Thanks for the links!
